superset-frontend/src/explore/actions/saveModalActions.js

```diff
@@ -43,6 +43,14 @@ export function fetchDashboards(userId) {
           value: id,
           label: (json.result[index] || {}).dashboard_title,
         }));
+        choices.sort((a, b) => {
+          const labelA = a.label.toUpperCase();
+          const labelB = b.label.toUpperCase();
+          return labelA.localeCompare(labelB, {
+            sensitivity: 'base',
+            numeric: true,
+          });
```

Comment:
`sensitivity: 'base'` already handles case sensitivity.

```suggestion
          return a.label.localeCompare(b.label, {
            sensitivity: 'base',
            numeric: true,
          });
```